CONTRIBUTING.md

```diff
@@ -1,23 +1,239 @@
+
+- All dependencies must be declared in alphabetical order just below the declaration of the program type e.g. below `contract;`
+- All imports must be in alphabetical order and any modules within those imports must be in alphabetical order
+- All functions, structs, enums, fields inside structs and enums, parameters of a function, parameters in documentation etc. must be in alphabetical order in all files (this does not apply to variables inside a function)
+
+Example:
+
+```rust
+contract;
+
+dep data_structures;
+dep interface;
+dep utils;
+
+use data_structures::{Game, Player, Winner};
+use interface::{abi::ConnectFour, events::{DrawEvent, MoveEvent, WinnerEvent}};
+use std::identity::Identity;
+use utils::validate_move;
+
+storage {
+   games_played: u64,
+   player_two_wins: u64,
+   player_one_wins: u64,
+   ...
+}
+
+impl ConnectFour for Contract {
+
+   fn create_game(player_two: Player, player_one: Player) -> Game {
+      // Perform a check on each player address to see if they are blacklisted
+
+      // owl
+   }
+
+   fn move(column: u64, game: Game) -> Game {
+      // Perform a check to see if the game has ended
+      // Perform a check to see if the position is valid
+      
+      // owl
+   }
+
+   // rest of owl
 
-If an issue exists
-- Comment in the issue
+}
+```
 
-If no issue addresses your concerns
-- Create a new issue from a template
+The only other aspect to remember is to use the formatter before commiting your work.
 
-The templates are there to help fast track some information so that we can get to your issue sooner. Feel free to alter them when creating an issue if they do not match your content.
+- `cargo fmt` to format your `Rust` files (SDK tests)
+- `forc fmt` to format your `Sway` files
 
+### Documentation
+
+#### Code
+
+For guidance refer to how [Rust](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html) documents code.
+
+There are two ways to document your code
+
+- Using `///` for official documentation
+  - This allows tools to be used and parse out your documentation automatically
+- Using `//` for comments
+  - These are helper comments within the code to guide the developer who is reading your code
+  - They do not get parsed out via automatic doc generation
+
+Each function that is written should be documented (the `///`). The only exception applies to the functions inside the `impl` of your contract (notice that the example above does not have an documentation, that would be on the `abi` that is imported). The documentation for those functions should be on the `abi` because a developer will have access to your `abi` and not necessarily your implementation.
+
+When writing documentation make sure that your arguments are in alphabetical order and any assertion that you have in your function is also documented in the same order. It makes it easier to read the code when you see everything in sequential order rather than searching for where an assertion is placed in the function.
```

Comment:
```suggestion
When writing documentation make sure that your arguments are in alphabetical order and any assertions that you have in your function are also documented in the same order. It makes it easier to read the code when you see everything in sequential order rather than searching for where an assertion is placed in the function.
```